tap_jira/http.py

```diff
@@ -59,6 +59,7 @@ def __init__(self, config):
             LOGGER.info("Using Basic Auth API authentication")
             self.base_url = config.get("base_url")
             self.auth = HTTPBasicAuth(config.get("username"), config.get("password"))
+            self.test_credentials_are_authorized()
```

Comment:
@savan-chovatiya, Looking at the code for 'test_credentials_are_authorized()', I think we should upgrade its code and add our standard raise_for_status() function. This function will help us to provide any helpful information to the customer for resolving the error

Reply:
@karanpanchal-crest , Our standard raise_for_status() function and proper exception handling is already added as part of #64